andela/ah-bird-box
# #163388261 feature user notifications

#### What does this PR do?
Have user notifications
#### Description of Task to be completed?
- write unit tests
- add email notification handler
- add email template
- add serializers and views
- add verbs
- add comment, article and follow handlers
#### How should this be manually tested?
1. Checkout to the branch ```ft-user-receive-notifications-163388261```
2. `pip install -r requirements.txt`
3. `python manage.py runserver`
4. On postman signup two users:POST url-`http://127.0.0.1:8000/api/users/`

    ```
            {
               "username":"yourusername",
		"email": "youremail",
		"password":"yourpassword"		
            }
    ```

Login the registered users.POST url-`http://127.0.0.1:8000/api/users/login/`

As a user, follow the other registered user. POST  url-`http://127.0.0.1:8000/api/profiles/username/follow/`

           **N.B:** the other usershould receive an email notification that he has been followed

As the other user, create an article . POST url-`http://127.0.0.1:8000/api/articles/`
  
    **N.B** notification should be sent to you that an article has been created

Method | Endpoint | Functionality
--- | --- | ---
GET | `http://127.0.0.1:8000/api/notifications + 'token'` | View all notifications
PUT | `http://127.0.0.1:8000/api/notifications/unsubscribe/ + 'token'` | Unsubscribe from notifications
POST | `http://127.0.0.1:8000/api/notifications/subscribe/ + 'token'` | Subscribe to notifications
POST | `http://127.0.0.1:8000/api/profiles/username/follow/ + 'token'` | Get email notification for follow
POST | `http://127.0.0.1:8000/api/articles/ + 'token'` | Get email notification for article created
POST | `http://127.0.0.1:8000/api/articles/slug/comments/ + 'token'` | Get email notification for comment on your article


#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388261](https://www.pivotaltracker.com/n/projects/2240063)
#### Screenshots (if appropriate)
followed notification
![image](https://user-images.githubusercontent.com/42655718/53207060-be01ab80-3643-11e9-819a-f47776e7c651.png)

comment notification
![image](https://user-images.githubusercontent.com/42655718/53207104-da9de380-3643-11e9-8079-fcf365f37a75.png)
get all notifications
![image](https://user-images.githubusercontent.com/42655718/53207291-6748a180-3644-11e9-83ff-f5990319aef8.png)

#### Questions:
